pkg/odo/cli/component/create.go

```diff
@@ -456,6 +579,27 @@ func (co *CreateOptions) Complete(name string, cmd *cobra.Command, args []string
 
 // Validate validates the create parameters
 func (co *CreateOptions) Validate() (err error) {
+	if experimental.IsExperimentalModeEnabled() {
+		if co.devfileMetadata.devfileSupport {
+			// Validate if the devfile component that user wants to create already exists
+			spinner := log.Spinner("Validating component")
+			defer spinner.End(false)
+
+			if util.CheckPathExists(DevfilePath) || util.CheckPathExists(EnvFilePath) {
```

Comment:
we need to check this before running `catalogDevfileList, err := catalog.ListDevfileComponents()`, currently if you run `odo create openLiberty` and then run `odo create openLiberty` in the same folder it is taking too much time to return `"Devfile.yaml or env.yaml already exists`, I think this should be the first validation.

Reply:
By design after we implement devfile support for odo, odo needs to support both devfile and s2i components, but devfile component has higher priority.

`catalogDevfileList, err := catalog.ListDevfileComponents()` helps us check devfile registry so that we can validate if the component is supported devfile component, if it's not supported devfile component we won't validate `Devfile.yaml or env.yaml already exists` and we go to the existing s2i code path. That's why we call `catalogDevfileList, err := catalog.ListDevfileComponents()` before `Devfile.yaml or env.yaml already exists`